client/src/main/java/com/netflix/conductor/client/task/WorkflowTaskCoordinator.java

```diff
@@ -23,426 +23,420 @@
+            this.taskWorkers = taskWorkers;
+            return this;
+        }
+
+        /**
+         * @param taskWorkers workers that will be used for polling work and task execution.
+         * @return Builder instance
+         */
+        public Builder withWorkers(Worker... taskWorkers) {
+            this.taskWorkers = Arrays.asList(taskWorkers);
+            return this;
+        }
+
+        /**
+         * Builds an instance of the WorkflowTaskCoordinator.
+         * <p>
+         * Please see {@link WorkflowTaskCoordinator#init()} method. The method must be called after this constructor
+         * for the polling to start.
+         * </p>
+         */
+        public WorkflowTaskCoordinator build() {
+            if (taskWorkers == null) {
+                throw new IllegalArgumentException(
+                    "No task workers are specified. use withWorkers() to add one or more task workers");
+            }
 
-	private ExecutorService executorService;
+            if (taskClient == null) {
+                throw new IllegalArgumentException("No TaskClient provided. use withTaskClient() to provide one");
+            }
+            return new WorkflowTaskCoordinator(eurekaClient, taskClient, threadCount, sleepWhenRetry, updateRetryCount,
+                taskWorkers, workerNamePrefix);
+        }
+    }
+
+    private final Thread.UncaughtExceptionHandler uncaughtExceptionHandler = (thread, error) -> {
+        // JVM may be in unstable state, try to send metrics then exit
+        WorkflowTaskMetrics.incrementUncaughtExceptionCount();
+        LOGGER.error("Uncaught exception. Thread {} will exit now", thread, error);
+    };
+
+    /**
+     * Starts the polling. Must be called after the constructor {@link #WorkflowTaskCoordinator(EurekaClient,
+     * TaskClient, int, int, int, Iterable, String)} or the builder {@link Builder#build()} method
+     */
+    public synchronized void init() {
+        if (threadCount == -1) {
+            threadCount = workers.size();
+        }
+
+        pollingSemaphore = new PollingSemaphore(threadCount);
+
+        LOGGER.info("Initialized the co-ordinator with {} threads", threadCount);
 
-	private ScheduledExecutorService scheduledExecutorService;
+        AtomicInteger count = new AtomicInteger(0);
 
-	private EurekaClient eurekaClient;
+        this.executorService = Executors.newFixedThreadPool(threadCount,
+            new BasicThreadFactory.Builder()
+                .namingPattern(workerNamePrefix + count.getAndIncrement())
+                .uncaughtExceptionHandler(uncaughtExceptionHandler)
+                .build());
+
+        this.scheduledExecutorService = Executors.newScheduledThreadPool(workers.size());
+        workers.forEach(worker -> scheduledExecutorService.scheduleWithFixedDelay(() -> pollForTask(worker),
+            worker.getPollingInterval(), worker.getPollingInterval(), TimeUnit.MILLISECONDS));
+    }
+
+    public void shutdown() {
```

Comment:
Javadoc on when and how it is invoked or should be invoked.

Reply:
added javadoc